### PR TITLE
feat: init block tag & database block

### LINKED ISSUE #633: Support Database
Upstream: https://github.com/toeverything/blocksuite/issues/629

### PR DESCRIPTION
**Pull request by himself65**
Upstream bug: https://github.com/toeverything/blocksuite/issues/648

Preview URL: https://blocksuite-git-20220110-tag-toeverything.vercel.app/?init=database

### CODE REVIEW

File: packages/global/src/index.d.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ declare namespace BlockSuiteInternal {
     ParagraphBlockModel,
     SurfaceBlockModel,
   } from '@blocksuite/blocks';
+  import TagBlockModel = BlockSuiteModelProps.TagBlockModel;
```
Comment (himself65) [inline packages/global/src/index.d.ts]: incorrect

**Review by doodlewind — COMMENTED** [review]
Let's have `BlockTag` rather than `TagBlock`, since tag itself should attach to blocks, and is not a type (but an attribute) of block.

**Comment by himself65** [comment]
rebased

**Comment by himself65** [comment]
This PR is no longer updated because we need to upgrade the infra. Please see the next PR

**Pull request closed without merging**